File: 2019/1/solution.py

def total_fuel_requirements(module_masses):
    """
    Puzzle part one solution
    """
    return sum([mass // 3 - 2 for mass in module_masses])

def main():
    """
    AOC 2019 Day 1 Driver
    """
    module_masses = None
    with open('input') as file:
        module_masses = [int(mass) for _, mass in enumerate(file)]

    print('Total fuel requirements for all modules on spacecraft:',
          total_fuel_requirements(module_masses))

if __name__ == '__main__':
    main()

File: 2019/1/solution.go
package main

import (
	"fmt"
	"io"
	"os"
)

const inputFilePath = "input"

// TotalFuelRequirements computes the sum of all fuel requirements for all
// modules based on their mass. Solves part one.
func TotalFuelRequirements(module_masses []int) (total int) {
	for _, mass := range module_masses {
		total += mass/3 - 2
	}
	return
}

// main driver for AOC 2019 #1
func main() {
	var module_masses []int

	file, err := os.Open(inputFilePath)
	if err != nil {
		panic(err)
	}
	defer file.Close()

	for {
		var mass int
		_, err = fmt.Fscanln(file, &mass)
		if err == io.EOF {
			break
		}
		if err != nil {
			panic(err)
		}

		module_masses = append(module_masses, mass)
	}

	fmt.Println("Total fuel requirements for all modules on spacecraft:",
		TotalFuelRequirements(module_masses))
}
